### PR TITLE
drivers: display: st7789v: Support 3-line serial interface

### PR DESCRIPTION
**Pull request by pdgendt**
The sitronix ST7789V serial interface can operate with 3- or 4-line protocol.

Tested on custom board.

### CODE REVIEW

File: drivers/display/Kconfig.st7789v [hunk 1]
```diff
@@ -9,6 +9,25 @@ menuconfig ST7789V
...
+	  contains just transmission byte and control bit D/CX is transferred
+	  by the D/CX pin.
+
+config ST7789V_SERIAL_3_LINE
+	bool "3-line"
+
+config ST7789V_SERIAL_4_LINE
+	bool "4-line"
+
+endchoice
```
Comment (gmarull) [inline drivers/display/Kconfig.st7789v:29]: this forces all instances to use one mode or the other, can this be moved to Devicetree instead?

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/display/Kconfig.st7789v (hunk 1 above)
Comment (pdgendt) [reply drivers/display/Kconfig.st7789v:29]: I thought so too, but the same is true for the pixel format, and I wanted to keep the changes to a minimum. A follow-up PR?

Comment (jfischer-no) [reply drivers/display/Kconfig.st7789v:29]: You would keep changes minimal if it used devicetree.

Comment (pdgendt) [reply drivers/display/Kconfig.st7789v]: I've removed the `Kconfig` option and reworked the logic to use an optional cmd/data pin. If it is not set, it operates in the 3-line serial protocol.

File: drivers/display/display_st7789v.c [hunk 2]
```diff
@@ -67,29 +68,42 @@ static void st7789v_set_lcd_margins(const struct device *dev,
...
+			spi_write_dt(&config->bus, &tx_bufs);
+		}
 
-	if (tx_data != NULL) {
-		tx_buf.buf = tx_data;
-		tx_buf.len = tx_count;
-		st7789v_set_cmd(dev, 0);
-		spi_write_dt(&config->bus, &tx_bufs);
+		if (tx_data != NULL) {
+			for (index = 0; index < tx_count; ++index) {
```
Comment (jfischer-no) [inline drivers/display/display_st7789v.c:102]: `		for (size_t index = 0; index < tx_count; ++index) {` ? since index is only used in the loop

**Review by gmarull — COMMENTED** [review]
could you please update bindings to mention 3-wire mode is enabled if cmd/data gpio is not provided?

**Comment by pdgendt** [comment]
> could you please update bindings to mention 3-wire mode is enabled if cmd/data gpio is not provided?

Done

**Review by gmarull — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**